tiledb/sm/group/group_directory.h

```diff
@@ -61,13 +61,10 @@ class GroupDirectory {
   /*     CONSTRUCTORS & DESTRUCTORS    */
   /* ********************************* */
 
-  /** Constructor */
-  GroupDirectory() = default;
```

Comment:
When there's no default constructor, it's better to explicitly delete it. Otherwise someone has to hunt for a possible definition.